Add detection for removed methods and kwargs (QKT101/102/201/202) (#2)
qiskit-community/flake8-qiskit-migration: README.md

# flake8-qiskit-migration

Flake8 plugin to detect deprecated/removed imports, methods, and arguments in Qiskit 1.0 and 2.0.

### Error codes

| Code | What it detects |
|------|-----------------|
| **QKT100** | Import paths deprecated in Qiskit 1.0 ([migration guide](https://docs.quantum.ibm.com/api/migration-guides/qiskit-1.0-features)) |
| **QKT101** | Method calls removed in Qiskit 1.0 (e.g. `.qasm()`, `.cnot()`, `.bind_parameters()`) |
| **QKT102** | Keyword arguments removed in Qiskit 1.0 (e.g. `PassManager.append(max_iteration=...)`) |
| **QKT200** | Import paths removed in Qiskit 2.0 ([migration guide](https://docs.quantum.ibm.com/migration-guides/qiskit-2.0)) |
| **QKT201** | Method calls removed in Qiskit 2.0 (e.g. `.c_if()`, `.add_calibration()`, `.drive_channel()`) |
| **QKT202** | Keyword arguments removed in Qiskit 2.0 (e.g. `transpile(backend_properties=...)`) |

> [!NOTE]
> QKT101/QKT102/QKT201 use heuristic detection (method name matching in files
> that import `qiskit`). Without type inference, false positives are possible
> but unlikely for the Qiskit-specific names we check. QKT202 tracks which

> > functions were imported from `qiskit` and has near-zero false positives.

Reviewer: ```suggestion
> functions were imported from `qiskit` and should have few false positives.
```
